src/operations/insert.ts

```diff
@@ -70,7 +70,7 @@ export class InsertOneOperation extends InsertOperation {
     super.execute(server, session, (err, res) => {
       if (err || res == null) return callback(err);
       if (res.code) return callback(new MongoServerError(res));
-      if (res.writeErrors) return callback(new MongoServerError(res.writeErrors[0]));
+      if (res.writeErrors) return callback(new MongoWriteConcernError(res.writeErrors[0]));
```

Comment:
Is this always correct? I don't think a writeError is always in the category of a WriteConcernError, but I could be wrong, lets sync to make sure this is correct

Reply:
Capturing what spoke about in person: This has been reverted to MongoServerError since not every write error is a write concern error. The ideal solution would be to make this a `WriteError` class, but that would call for some error hierarchy refactoring since `WriteError` isn't a child of anything and that would be out of scope.